server/data/data-cassandra/src/main/java/org/apache/james/rrt/cassandra/CassandraRecipientRewriteTable.java

```diff
@@ -63,7 +65,13 @@ public Mappings getStoredMappings(MappingSource source) {
 
     @Override
     public Map<MappingSource, Mappings> getAllMappings() {
-        return cassandraRecipientRewriteTableDAO.getAllMappings().block();
+        return cassandraRecipientRewriteTableDAO.getAllMappings()
+            .map(pair -> new UserMapping(pair.getLeft(), pair.getRight()))
```

Comment:
Can't we get ride of that intermediate `UserMapping` representation that do not seem to add value?